core/src/main/java/org/frankframework/http/openapi/OpenApiGenerator.java

```diff
@@ -0,0 +1,269 @@
+		JsonObjectBuilder root = Json.createObjectBuilder();
+		root.add("openapi", "3.0.0");
+		String instanceName = AppConstants.getInstance().getProperty("instance.name");
+		String environment = AppConstants.getInstance().getString("dtap.stage", "LOC");
+		JsonObjectBuilder info = Json.createObjectBuilder();
+		info.add("title", instanceName);
+		info.add("description", "OpenApi auto-generated at " + DateFormatUtils.getTimeStamp() + " for " + instanceName + " (" + environment + ")");
+		info.add("version", "unknown");
+		root.add("info", info);
+		root.add("servers", mapServers(endpoint));
+
+		JsonObjectBuilder paths = Json.createObjectBuilder();
+		JsonObjectBuilder schemas = Json.createObjectBuilder();
+
+		for (ApiDispatchConfig config : clients) {
+			JsonObjectBuilder methods = Json.createObjectBuilder();
+			ApiListener listener = null;
+			for (ApiListener.HttpMethod method : config.getMethods()) {
+				JsonObjectBuilder methodBuilder = Json.createObjectBuilder();
+				listener = config.getApiListener(method);
+				if (listener != null && listener.getReceiver() != null) {
+					IAdapter adapter = listener.getReceiver().getAdapter();
+					if (StringUtils.isNotEmpty(adapter.getDescription())) {
+						methodBuilder.add("summary", adapter.getDescription());
+					}
+					if (StringUtils.isNotEmpty(listener.getOperationId())) {
+						methodBuilder.add("operationId", listener.getOperationId());
+					}
+					// GET and DELETE methods cannot have a requestBody according to the specs.
+					if (method != ApiListener.HttpMethod.GET && method != ApiListener.HttpMethod.DELETE) {
+						mapRequest(adapter, listener.getConsumes(), methodBuilder);
+					}
+					mapParamsInRequest(adapter, listener, methodBuilder);
+
+					methodBuilder.add("responses", mapResponses(adapter, listener.getContentType(), schemas));
+				}
+				methods.add(method.name().toLowerCase(), methodBuilder);
+			}
+			if (listener != null) {
+				paths.add(listener.getUriPattern(), methods);
+			}
+		}
+		root.add("paths", paths.build());
+		JsonObjectBuilder components = Json.createObjectBuilder();
+		components.add("schemas", schemas);
+		root.add("components", components);
+
+		return root.build();
+	}
+
+	private static JsonArrayBuilder mapServers(String endpoint) {
+		JsonArrayBuilder serversArray = Json.createArrayBuilder();
+		String servletPath = AppConstants.getInstance().getString("servlet.ApiListenerServlet.urlMapping", "/api");
+
+		// Get load balancer.url if exists
+		String loadBalancerUrl = AppConstants.getInstance().getProperty("loadBalancer.url", null);
+		if (StringUtils.isNotEmpty(loadBalancerUrl)) {
+			serversArray.add(Json.createObjectBuilder().add("url", loadBalancerUrl + servletPath).add("description", "load balancer"));
+		} else if (StringUtils.isNotBlank(endpoint)) { // fall back to the request endpoint
+			serversArray.add(Json.createObjectBuilder().add("url", endpoint));
+		}
+		return serversArray;
+	}
```

Comment:
Deze code is het zelfde gebleven (of zie ik het fout?), ik dacht dat hier een fout in zat.

Reply:
Nee, dat bleek toch goed te zijn. Zie description bovenaan de PR. Anders effe bellen als je er niet mee eens bent? 
Ik dacht dat eerst wel, maar later begreep ik van je dat als de request er niet is, de `loadBalancer.url` opgepakt moet worden en dat werkt.